subsys/net/ip/icmp.c

```diff
@@ -358,6 +384,45 @@ static int send_icmpv6_echo_request(struct net_icmp_ctx *ctx,
 }
 #endif
 
+static int get_offloaded_ping_handler(struct net_if *iface,
+				      net_icmp_offload_ping_handler_t *ping_handler)
+{
+	struct net_icmp_offload *offload;
+	int ret;
+
+	if (!IS_ENABLED(CONFIG_NET_OFFLOAD)) {
+		return -ENOTSUP;
+	}
+
+	if (iface == NULL) {
+		return -EINVAL;
+	}
+
+	if (!net_if_is_ip_offloaded(iface)) {
```

Comment:
We have a helper function that covers both offloading scenarios (https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/net/ip/net_if.c#L4247), perhaps we should make it public and use it instead?

Reply:
Yes, makes sense, will do.